### DISCUSSION TITLE
Clear Data On Exit

### DISCUSSION THREAD

**Discussion by blogmangit** [bromite/bromite#2477] [Ideas]
This feature is essential for those who want to use bromite as normal mode. We can use bromite in always incognito mode but the problem is in that case the links do not open in external app instead it opens in bromite incognito tab. So I think there should be an option to clear data on exit for normal mode users just like brave or vivaldi or edge provides us with one.

**Comment by bat999** [comment]
> This feature is essential for those who want to use bromite as normal mode.
So I think there should be an option to clear data on exit for normal mode users...
>
You need to open a New Issue Feature Request for that to happen.

**Reply by blogmangit** [reply]
I know but to my surprise a lots of users have opened issues regarding this including me.

**Reply by bat999** [reply]
I can see your feature request is marked as Duplicate.
https://github.com/bromite/bromite/issues/1525
There are several other posts about this subject.
This reply from uazo looks interesting, but it's Duplicate and Closed.
"kSavingBrowserHistoryDisabled that inhibits the saving of the history"
https://github.com/bromite/bromite/issues/1735
But maybe it's more complicated than it looks.

**Reply by uazo** [reply]
> "kSavingBrowserHistoryDisabled that inhibits the saving of the history"

has already been implemented:
![image](https://user-images.githubusercontent.com/29201891/205496343-c6b655bf-8b25-4081-818a-36269bbf797e.png)

**Reply by bat999** [reply]
Heck, we're gonna need a better constant!
One that...
Allows storing history during session
And
Inhibits saving history after session
:-)

**Reply by uazo** [reply]
> Inhibits saving history after session

do you mean maybe save the history in memory? why? I don't understand the objective.

**Reply by bat999** [reply]
So that browsing data (eg recently visited and cookies etc) is available during the session.
As if Expire history days was NOT set to No history.
Then delete it automatically on exit.

**Reply by uazo** [reply]
we probably don't understand each other, my fault and the fault of my English.

browsing data is **always** available during the session, and it is different from history, of course.

so, you want **all** the data not maintained? then browse incognito. 
you want **only** the history not maintained? then act with that flag.
do you want the history active but automatically deleted? then the only option is to keep it in memory. but **why** do you need it?

**Reply by bat999** [reply]
do you want the history active but automatically deleted?
Yes

 but **why** do you need it?
So that we can go back to a previous website that we have closed.

Maybe for Cookies and site data too.
I'm not sure about that, but with No history maybe we will have to accept cookies again if we return to a website during session.

Maybe Saved passwords, Auto-fill form data, Site settings too.

**Reply by uazo** [reply]
so your request is an always incognito with history active but in memory?

> Maybe Saved passwords, Auto-fill form data, Site settings too.

it is already possible to activate them in the always incognito.

**Reply by bat999** [reply]
Hi
Without Incognito Mode.
Without No history setting.

During a session we collect browsing data...
Browsing History
Cookies and site data
Saved passwords
Auto-fill form data
Site settings

An option in Settings -> Privacy and security "Delete browsing data on exit".

This feature has been suggested many times...
https://github.com/bromite/bromite/issues/426
It would need yet another patch to be developed and maintained.

**Reply by blogmangit** [reply]
No one here is trying to understand the fact that in only incognito mode serves the purpose but don't allow to open links in respective apps or respective external apps. The links always open in bromite.

For eg youtube links don't open in youtube app instead they open in bromite. That's the only problem of activating incognito. 

I don't understand why no one is accepting this fact or not trying to understand it.

**Reply by Universalizer** [reply]
> I know, as a one man army our dev doing Himalayan tasks.

Now he's not alone, don't forget, @uazo is collaborating him genuinely, thanks to his efforts.

**Comment by Universalizer** [comment]
### If this is the matter (For better understanding) ?

**Menu** > Setting > Privacy and security > Clear browsing data > last known selected options > **Clear data**

"Directly to execute while on clicking Exit but especially in normal mode" browsing session (not related to Incognito mode)

Other way
**Menu** > History > Clear browsing data > last known selected options > **Clear data, (Directly on Exit) but in normal browsing mode.**
